### PR TITLE
use wagmi hooks

### PR DESCRIPTION
**Pull request by cctdaniel**
- add copy button to code blocks
- make pre overflow auto
- fix response padding
- add ConnectKit wallet adapter
- refactor code to use wagmi hooks (we dont have to manually keep track and add rpc urls and config in the future since wagmi provides them)
- keep client `networkName` in sync when network is switched from ConnectKit wallet
- wagmi hook required proper typing so had to change tsconfig to use strict and also fix type errors for other components

### CODE REVIEW

**Review by jayantk — COMMENTED** [review]
I think there may be an issue with walletconnect as the vercel preview doesn't seem to work (I sent you a slack message).

File: __tests__/validateCodeSnippets.ts [hunk 1]
```diff
@@ -120,8 +120,8 @@ function validateCodeSnippets(directoryPath: string): void {
     } else if (stats.isFile() && file.endsWith(".mdx")) {
       const codeSnippets = extractCodeSnippetsFromFile(fullPath);
       codeSnippets.forEach((snippet, index) => {
-        const language = snippet.split("\n")[0].trim();
-        const code = snippet.slice(language.length).trim();
+        const language = snippet.split("\n")[0].split(" ")[0].trim();
+        const code = snippet.slice(snippet.split("\n")[0].trim().length).trim();
```
[the remark this answers is not in the document]
Comment (cctdaniel) [reply __tests__/validateCodeSnippets.ts:124]: because I added `copy` and `language.length` does not take into consideration of `copy`

**Review by jayantk — APPROVED** [review]

File: __tests__/validateCodeSnippets.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (cctdaniel) [reply __tests__/validateCodeSnippets.ts]: I added a variable `firstLine` to make it clearer

File: components/Example.tsx [hunk 2]
```diff
@@ -66,7 +85,13 @@ export class ExampleRenderingContext {
   // Get the price feed id for the provided symbol name (e.g., "Crypto.BTC/USD").
   // This function will automatically account for different networks.
   public getFeedId(symbolName: string): string {
-    return KnownFeedIds[this.networkType][symbolName];
+    const feedIds = KnownFeedIds[this.networkType] as { [key: string]: string };
+    if (feedIds) {
```
[the remark this answers is not in the document]
Comment (cctdaniel) [reply components/Example.tsx:89]: ah yes good catch

**Pull request merged**